### DISCUSSION TITLE
"No username received in stomp message"

### DISCUSSION THREAD

**Discussion by AAndersn** [GRIDAPPSD/gridappsd-forum#127] [API] [answered]
@poorva1209 @craig8 

I was recently asked why there frequently warnings `No username received in stomp message` when authenticating via GridAPPSD-Python

Is this an expected behavior? If so, what is the reason this message gets printed?

```
2022-05-23 16:24:30,067 Thread-89 WARN [pnnl.goss.core.client.GossClient] - No username received in stomp message
2022-05-23 16:24:32,086 Thread-138 INFO [gov.pnnl.goss.gridappsd.log.LogManagerImpl] - 1653323072081|gov.pnnl.goss.gridappsd.process.ProcessEvent|1251656980|RUNNING|system|INFO
New process id generated with new process type
```

**Comment by poorva1209** [comment, marked as answer]
This warning message is published when a user is trying to make a new connection using `gad=GridAPPSD() `. The connection requests a unique token for authentication and authorization with a message shown below. Since the security check is not completed at this step, the system is throwing a warning message. But you are right the message is misleading from a user's perspective and we can change it. 

```
MESSAGE
content-length:20
subscription:2
expires:0
destination:/topic/pnnl.goss.token.topic
timestamp:1653324489606
message-id:ID\c05373756cd14-33005-1653089008786-4\c162\c-1\c1\c1reply-to:temp.token_resp.system-2022-05-23 09\c48\c08.467433
priority:4
```

@tdtalbot  I am creating an issue for this and assigning it to you. Let me know if you have any questions.

**Comment by AAndersn** [comment]
@arunveeramany, please see Poorva's response above.
